lab_4_retrieval_w_clustering/main.py

```diff
@@ -287,6 +398,26 @@ def run_single_train_iteration(self) -> list[ClusterDTO]:
         Returns:
             list[ClusterDTO]: List of clusters.
         """
+        centroids = []
+        for cluster in self.__clusters:
```

Comment:
Are you sure that this method should not return a new clusters? 
You set new clusters into self.__clusters, but I guess you need to create new and return them. And then use it in `train` as an input for `self._is_convergence_reached`

Reply:
agree. i changed this method so now it returns a list with some new clusters (not self.__clusters), and because of this i also changed `train`. can you take a look at it, please?